python/lsst/ts/ess/csc/ess_csc.py

```diff
@@ -165,22 +168,23 @@ async def process_temperature_telemetry(
         timestamp = data[1]
         sensor_data = data[3:]
         device_configuration = self.device_configurations[sensor_name]
-        telemetry = {"sensorName": sensor_name, "timestamp": timestamp}
+        telemetry = {
+            "sensorName": sensor_name,
+            "timestamp": timestamp,
+            "numChannels": device_configuration.num_channels,
+        }
 
         if len(sensor_data) != device_configuration.num_channels:
             raise RuntimeError(
                 f"Expected {device_configuration.num_channels} temperatures "
                 f"but received {len(sensor_data)}."
             )
-        for i, value in enumerate(sensor_data):
-            # The telemetry channels start counting at 1 and not 0.
-            telemetry[f"temperatureC{i + 1:02d}"] = value
-        self.log.info(f"Received temperatures {telemetry}")
-        self.log.info("Sending telemetry.")
-        telemetry_method = getattr(
-            self, f"tel_temperature{device_configuration.num_channels}Ch"
-        )
-        telemetry_method.set_put(**telemetry)
+        temperature = np.zeros(NUMBER_OF_TEMP_CHANNELS)
+        for i in range(len(sensor_data)):
+            temperature[i] = sensor_data[i]
```

Comment:
It is more efficient to use subrange indexing, e.g.:
```
temperature[: device_configuration.num_channels] = sensor_data
```
You might need [:] after sensor_data, but I doubt it.